README.md

```diff
@@ -0,0 +1,129 @@
+# Jsonapi Orchestrator
+
+Building better jsonapi-compliant code
+
+[![BCH compliance](https://bettercodehub.com/edge/badge/MyJobGlasses/jsonapi-orchestrator?branch=master)](https://bettercodehub.com/)
+
+[![CircleCI](https://circleci.com/gh/MyJobGlasses/jsonapi-orchestrator.svg?style=svg)](https://circleci.com/gh/MyJobGlasses/jsonapi-orchestrator)
+
+# What is it ?
+
+Ever wanted to use...
+- json:api communication with your APIs
+- Redux based storage of resources retrieved from the API
+- Side effects with redux-saga posting to your APIs
+- Caching expensive requests to alleviate load on your APIs
+
+**Jsonapi Orchestrator is For You 🎁 🎉**
+
+# HOW TO
+
+Jsonapi Orchestrator helps you to deal with various stages of json:api based API interaction
+
+## Building json:api Requests
+
+A json:api request can be categorized either as a READ request (GET request) or a WRITE request (POST/PATCH/PUT).
+
+Building those requests can be difficult because of the format of filters, sorting, and includes, in addition to specifying the endpoint. In addition, if you ever want to handle caching of requests, you would need to be ablte to specify metadata such as data freshness to decide later if you actually want to fire the query or reuse existing data already fetched.
+
+Our Jsonapi builders will let you store this metadata so it can be reused with cache managers.
+
+Here are some examples of basic read and writes
+
+### APIs
+
+You can manage a list of multiple APIs easily
+
+```javascript
+// my-apis.js
+const EMPLOYEE_API_V1 = new API({
+  name: 'Employee API v1', url: 'https://employee.example.com/api/v1'
+})
+const EMPLOYEE_API_V2 = new API({
+  name: 'Employee API v2', url: process.env.EMPLOYEE_API_V1_URL
+})
+const METRICS_API_V2 = new API({
+  name: 'Metrics API v2', url: 'https://metrics.example.com/api/v2'
+})
+
+export default { METRICS_API_V2, EMPLOYEE_API_V1, EMPLOYEE_API_V2 }
+
+// somewhere
+import APIs from 'my-apis'
+```
+
+### Basic READ of single document
+
+```javascript
+employeeReader = new JsonapiResourceReader({ type: 'employee'})
+employeeReader.sideload({ company: { admins: true } })
+employeeReader.filter({ in_vacation: false })
+employeeReader.sort({ next_holiday_at: 'asc' })
+
+requestBuilder = new JsonapiRequestBuilder({
+  resource: employeeReader,
+  method: 'GET',
```

Comment:
We need to specify method param ? Why not just use `instanceof` check on resource to determine which method to use ?

Reply:
seem to be coded btw

Reply:
yep wrote this a month ago, needs cleanup